features/machine/machine.feature

```diff
@@ -141,4 +141,36 @@ Feature: Machine features testing
     Given I scale the machineset to +2
     Then the machineset should have expected number of running machines
 
-    
+  # @author miyadav@redhat.com
+  # @case_id OCP-24363
+  @admin
+  @destructive
+  Scenario: [MAO] Reconciling machine taints with nodes
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+
+    Given I store the number of machines in the :num_to_restore clipboard
+    And admin ensures node number is restored to "<%= cb.num_to_restore %>" after scenario
+
+    Given I clone a machineset named "machineset-24363"
```

Comment:
Sorry If I'm missing the thread here. We create a machine set, do we remove it afterwards? Isn;t it easier to query which machines are part of it instead of ` I store the last provisioned machine in the :machine clipboard`. The scenario is not easy to follow why are we doing things.

If I understand correctly we want something like:
```
    Given I have an IPI deployment
    And I switch to cluster admin pseudo user

    Given admin ensures machine_set "machineset-24363" is deleted after scenario
    And I clone a machineset named "machineset-24363"
    And evaluation of `machine_set.machines.first.node_name` is stored in the :noderef_name clipboard

    ... (do the patching and check as before)
```

It may require slight adjustments to machine_set class perhaps to work as above. But would make scenario easier to follow (given I actually understand what it does now).

Reply:
The `clone` step auto registers `admin ensures deleted` in after scenario.

Reply:
Yes, I'd love to do that(implement a machines method for machine_set class). The machineset resource does not include any machine information, to know which machine belongs to a machineset, we must query all the machines and filter them by their `machine.openshift.io/cluster-api-machineset` label. So in order to implement a `machines` method for the machineset, I need to find a way to list all machines inside the mahcineset class.

@akostadinov Could you instruct how can I implement it? Thanks

Reply:
@jhou1  @akostadinov  made the changes to incorporate :

Validated looking good -
        Type    Reason                   Age                  From                                                 Message
        ----    ------                   ----                 ----                                                 -------
        Normal  NodeHasSufficientMemory  106s (x8 over 118s)  kubelet, ip-10-0-140-185.us-east-2.compute.internal  Node ip-10-0-140-185.us-east-2.compute.internal status is now: NodeHasSufficientMemory
        Normal  NodeHasNoDiskPressure    106s (x8 over 118s)  kubelet, ip-10-0-140-185.us-east-2.compute.internal  Node ip-10-0-140-185.us-east-2.compute.internal status is now: NodeHasNoDiskPressure
      
      [08:21:01] INFO> Exit Status: 0
      | resource | node                   |
      | name     | <%= cb.noderef_name %> |
    Then the output should contain:                                                                   # features/step_definitions/common.rb:33
      | role=master:NoExecute  |
      | role=master:NoSchedule |
waiting for operation up to 3600 seconds..
waiting for operation up to 300 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
      [08:21:01] INFO> === After Scenario: [MAO] Reconciling machine taints with nodes ===
      [08:21:01] INFO> Shell Commands: oc delete machinesets machineset-24363 --wait=false --config=/root/workdir/c92ee1977c9c-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
      machineset.machine.openshift.io "machineset-24363" deleted
      
      [08:21:03] INFO> Exit Status: 0
      [08:21:05] INFO> oc get machinesets machineset-24363 --output=yaml --config=/root/workdir/c92ee1977c9c-/ocp4_admin.kubeconfig --namespace=openshift-machine-api
      [08:21:05] INFO> After 1 iterations and 2 seconds:
      
      STDERR:
      Error from server (NotFound): machinesets.machine.openshift.io "machineset-24363" not found
      
      [08:21:05] INFO> Shell Commands: rm -r -f -- /root/workdir/c92ee1977c9c-
      
      [08:21:05] INFO> Exit Status: 0
      [08:21:05] INFO> === End After Scenario: [MAO] Reconciling machine taints with nodes ===

1 scenario (1 passed)
10 steps (10 passed)
5m26.217s
[08:21:05] INFO> === At Exit ===

Reply:
successful v3 run link after the change - https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/85624/console